.github/workflows/check-parse-results.yml

```diff
@@ -0,0 +1,28 @@
+name: Check parse results
+on:
+  workflow_call:
+  workflow_dispatch:
+jobs:
+  check-parse-results:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          cache: yarn
+      - run: yarn install --frozen-lockfile
+      - run: yarn build
+      - uses: actions/checkout@v3
+        with:
+          repository: DefinitelyTyped/DefinitelyTyped
+          path: DefinitelyTyped
+          # So we can diff the most recent commit and its parent.
+          # https://github.com/microsoft/types-publisher/pull/290
+          fetch-depth: 2
+      - name: Check parse results
+        run: node --require source-map-support/register packages/dtslint-runner/ --path DefinitelyTyped/
```

Comment:
dtslint-runner isn't the same as check-parse-results in definitionParser, is it?

Reply:
@sandersn Yeah, these days check parse results is only called from dtslint-runner:
```sh
git grep --files-with-matches checkParseResults
packages/definitions-parser/src/check-parse-results.ts
packages/dtslint-runner/src/prepareAffectedPackages.ts
packages/dtslint-runner/src/prepareAllPackages.ts
```

That's probably alright?

The workflow is meant to confirm check parse results and the DT repo are in sync, to avoid DT CI outages. This PR would've cause `checkParseResults()` (and every DT PR except https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60857#issue-1275806812) to fail if we hadn't fixed up the DT repo first. Before the DT half of this PR landed, the workflow showed the DT repo was out of sync: https://github.com/microsoft/DefinitelyTyped-tools/runs/6970784854?check_suite_focus=true

Reply:
Will PRs change the result of checkParseResults? I think it's only DT PRs that will, in which case the right place for the check to run is dtslint-runner, both the CI version and the overnight version.

Reply:
@sandersn Well, changes to `getTypingInfo()`/`getModuleInfo()` change the dependency graph (potentially) which changes `checkParseResults()` and the DT repo states it accepts/rejects. That's the case with this PR.

Whether it happens that often, maybe not. But it's inexpensive to check, I think, and much nicer than finding out in the DT CI, after the `@definitelytyped` tools are deployed to npm and every DT PR fails?

Whereas the unit tests confirm the implementation matches assumptions, actually running check parse results confirms the assumptions match reality (at least the ones we can quickly check, and that potentially impact every PR). I think it's worth it?